### PR TITLE
feat(discover): Add ability to create saved query

### PR DESCRIPTION
**Pull request by lynnagara**

### CODE REVIEW

File: src/sentry/api/endpoints/organization_discover_query.py [hunk 1]
```diff
@@ -13,14 +13,20 @@
...
 from sentry.api.bases import OrganizationEndpoint
 from sentry.models import Project, ProjectStatus, OrganizationMember, OrganizationMemberTeam
 from sentry.utils import snuba
 from sentry import roles
 from sentry import features
 
 
+class OrganizationDiscoverQueryPermission(OrganizationPermission):
+    scope_map = {
+        'POST': ['org:read', 'project:read'],
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_discover_query.py]: Is there no GET now?

File: src/sentry/api/endpoints/organization_discover_saved_queries.py [hunk 2]
```diff
@@ -1,17 +1,65 @@
...
+        allow_null=True,
+    )
+
+    def validate(self, data):
+        return {
+            'name': data['name'],
+            'project_ids': data['projects'],
+            'query': {
+            }
+        }
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_discover_saved_queries.py]: Is there more coming here?

File: tests/snuba/test_organization_discover_saved_queries.py [hunk 3]
```diff
@@ -36,3 +36,20 @@ def test_get(self):
...
+                'fields': [],
+                'range': '24h',
+                'limit': 20,
+                'conditions': [],
+                'aggregations': [],
+                'orderby': '-time',
+            })
+
+        assert response.status_code == 201, response.content
+        assert response.data['name'] == 'New query'
```
Comment (markstory) [inline tests/snuba/test_organization_discover_saved_queries.py]: Should you check any other parts of the query were persisted/part of the response?

File: src/sentry/api/endpoints/organization_discover_query.py (hunk 1 above)
Comment (lynnagara) [reply src/sentry/api/endpoints/organization_discover_query.py]: I realised the other endpoint can use standard OrganizationPermission, this one is just specific to discover queries (post)

File: tests/snuba/test_organization_discover_saved_queries.py (hunk 3 above)
Comment (lynnagara) [reply tests/snuba/test_organization_discover_saved_queries.py]: 👍 added assertions

File: src/sentry/api/endpoints/organization_discover_saved_queries.py (hunk 2 above)
Comment (lynnagara) [reply src/sentry/api/endpoints/organization_discover_saved_queries.py]: Oops, yes, pushed the rest of the changes

**Review by markstory — APPROVED** [review]

**Pull request merged**